### ISSUE TITLE
Remove remote url before save

### ISSUE THREAD

**Issue by brijeshror** [carrierwaveuploader/carrierwave#2067]
How can I remove remote url after assign a url? I tried following things but not working
`[40] pry(main)> obj.remote_image_url_url = "http://url"`
`[40] pry(main)> obj.remote_image_url_url = "http://url"`
`=> "http://url"`
`[41] pry(main)> obj.remote_image_url_url = nil`
`=> nil`
`[42] pry(main)> obj.remote_image_url_url`
`=> "http://url"`

I also tried with obj.remote_image_url_url.clear but still I am getting error on save
`=> ["Image url could not download file: 403 Forbidden"]`

Basically I am checking on save that if there is any error on remote url then image url will be set to nil and record should be saved.

**Comment by rogerluo410** [comment]
Same issue,  I even think the remote url feature is not useful at all. 

I want to know how could I unset remote_xxx_url in carrierwave, like this question asks:

**Comment by jonbuda** [comment]
Still an issue in Carrierwave 2.02. there's no way to clear out the `remote_url` if it fails an integrity check or validation. I had to do this (which I hate) but it's the only way I got it working. not ideal.

```
obj.remove_image!
obj.remote_image_url = nil
obj.image = nil
obj.instance_variable_get(:@_mounters)[:image].instance_variable_set(:@remote_urls, [])
obj.instance_variable_get(:@_mounters)[:image].integrity_errors.clear
obj.save!
```

**Comment by pjg** [comment]
For anyone stumbling into this issue, here's how I solved it. Since `remote_image_url` cannot be cleared, when user is being created I am excluding the `remote_image_url`, then changing it to be an url, and in case of an error, doing `user.reload` to clear the `remote_image_url`.

```ruby
def create_user
  user = User.new user_params.except(:remote_image_url)
  user.save!
end

def update_image
  user.remote_image_url = ...
  save!
rescue ActiveRecord::RecordInvalid
  user.reload
end
